cmd/internal/kube/inject.go

```diff
@@ -28,13 +32,41 @@ var (
 	// When set to "true", injectCmd will prepend a secret to each injectable namespace found in the file to inject (injectFileNameFlag)
 	// Otherwise, injectCmd will treat secretInjectFlag as the file path all secrets should be generated to
 	secretInjectFlag string
+
+	// Postman related flags
+	postmanCollectionID string
+	postmanEnvironment  string
 )
 
 var injectCmd = &cobra.Command{
 	Use:   "inject",
 	Short: "Inject Akita into a Kubernetes deployment",
 	Long:  "Inject Akita into a Kubernetes deployment or set of deployments, and output the result to stdout or a file",
 	RunE: func(_ *cobra.Command, args []string) error {
+		if projectNameFlag == "" && postmanCollectionID == "" {
+			return cmderr.AkitaErr{
+				Err: errors.New("exactly one of --project or --collection must be specified"),
+			}
+		}
+
+		// If --collection was given, set rest.domain and environment config (if given)
+		if postmanCollectionID != "" {
+			if postmanEnvironment != "" {
+				env := strings.ToUpper(postmanEnvironment)
+
+				cfg.WritePostmanEnvironment("default", env)
+			}
+
+			rest.Domain = "staging.akita.software"
```

Comment:
I am a bit worried about this.

What I would like is to change the _default_ domain based on the postman environment. What you've done here will overwrite the user's --domain flag even if they gave it explicitly.

I don't think this is something that the flags library makes easy.  We probably have to use "" as the default domain for the flag, so we can tell whether it's been set or not.  (Unless there's a way to query the flag?)

```
        rootCmd.PersistentFlags().StringVar(&rest.Domain, "domain", defaultDomain, "Your assigned Akita domain (e.g. company.akita.software)")
        rootCmd.PersistentFlags().MarkHidden("domain")
```

We probably also need to make the environment flag a root command flag.  Then in `preRun` we can examine the environment flag and the rest domain flag and set rest.Domain appropriately.  This needs to happen before the call to telemetry.Init(), which will make an API call.

https://github.com/akitasoftware/akita-cli/blob/890c2fbfe21c506453cfda526c8d3b55085d0e1e/cmd/root.go#L76C7-L77C22

However, I don't think this matters for the `kube` commands which as far as we know don't make API calls at all, except for the telemetry setup?

Reply:
Ah, I see this is not true because you introduced a call to `lookupService` below.

Reply:
As discussed will be removing the `environment` flag and will pick domain setting in a separate task for next milestone.